### PR TITLE
RoR project - Social Media Website - Friendships v1

### PR DESCRIPTION
**Pull request by mrigorir**
This is the Second milestone from a forked project [microverseinc/ror-social-scaffold](https://github.com/microverseinc/ror-social-scaffoldurl) 

and its goal is to prepare microverse students for a social website app, using associations, 

table-relations, ERD's, user validations, and MVP concepts to put our knowledge into practice.

I have created associations, methods, and features for the Friendship model for this milestone, with is directly bonded to the User model.

### CODE REVIEW

File: config/database.yml [hunk 1]
```diff
@@ -17,14 +17,17 @@
 default: &default
   adapter: postgresql
   encoding: unicode
+  database: social_media_app
+  username: postgres
+  password: password
```
Comment (eliasecasta) [inline config/database.yml]: Don't push this 2 values

**Review by eliasecasta — CHANGES_REQUESTED** [review]
# Status:  Changes Required ♻️

Hello @mrigorir,

Nice job done so far!

I think that you missed a really important part of this milestone, let me give you more detail about the missing features:

- Currently, all created accounts are instantly friends to each other, you missed completely the friend requests, approvals, and unfriending features please add them because they are mandatory for this project.
- - [x] User can invite friends ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/send_invitation.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
- - [x] User can see friendships requests ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/pending_request.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
- - [x] User has to accept friendship request to start friendship ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/accept_reject_friendship.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
- - [x] User can reject a friendship request ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/accept_reject_friendship.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
- - [x] There is a button to invite user to friendship on users list page and on single user page ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/user_can_see_invite_buttton.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
- - [x] Posts timeline includes posts written by logged-in user and their friends ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/)) (Currently this works because all of them all already friends, but what happens if they are not? Will they disappear?)
- - [x] All your tests are failing for me but it might be a problem from my side, please double-check that all your tests are passing and if they are, ignore this comment.
- - [x] Remove your DB credentials in your database.yml, this is to reinforce security and good habits of not pushing security data to the public.

Please correct those issues and send another review request, good luck, and well done!

**Review by cyonii — CHANGES_REQUESTED** [review]
### STATUS: CHANGES ♻️✍️

Hello @mrigorir, you have done a really great job implementing the previously requested changes. Way to go!


#### TO IMPROVE
You have missed one mandatory feature, kindly check below and fix these issues before resubmitting for review
- [x] Posts timeline should include posts written by logged-in user and their confirmed friends ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/)). Currently I can see every posts including posts of non-friends.
```ruby
# User model

def friends_and_own_posts
  Post.where(user: (self.friends.to_a << self))
  # This will produce SQL query with IN. Something like: select * from posts where user_id IN (1,45,874,43);
end
```

> #### <br>You can ping me on Slack @[CY Kalu](https://microverse-students.slack.com/team/U01494U96FR) to draw my attention in case I do not respond quickly on here. I'm always available and will be glad to help.<br><br>

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

### Hi @mrigorir ! :wave: :wave:

####  Good job for work making all the necessary changes requested by the previous code reviewer(s). You are ready to proceed to the next milestone! 👍  Congratulations!  :clap:


#### You may merge, good luck on your next milestone! :rocket:

If you have any queries, please feel free to drop me a comment 😉.

#### Happy Coding! :+1:

__Robert Loterh __
[GitHub](https://github.com/rloterh) | [LinkedIn](https://www.linkedin.com/in/rloterh/) | [Twitter](https://twitter.com/RLoterh)

**Pull request merged**